components/announcement_bar/freemium_trial_ended_announcement_bar/index.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {t} from 'utils/i18n';
+
+import AnnouncementBar from '../default_announcement_bar';
+import {AnnouncementBarTypes, Preferences, CloudBanners} from 'utils/constants';
+import useGetUsageDeltas from 'components/common/hooks/useGetUsageDeltas';
+import useGetLimits from 'components/common/hooks/useGetLimits';
+import {GlobalState} from 'types/store';
+import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import useGetSubscription from 'components/common/hooks/useGetSubscription';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {
+    getCurrentUser,
+} from 'mattermost-redux/selectors/entities/users';
+import {anyUsageDeltaValueIsNegative} from 'utils/utils';
+import './freemium_trial_ended_announcement_bar.scss';
+
+const CloudFreemiumTrialEndAnnouncementBar: React.FC = () => {
+    const usageDeltas = useGetUsageDeltas();
+    const limits = useGetLimits();
+    const subscription = useGetSubscription();
+    const getCategory = makeGetCategory();
+    const dispatch = useDispatch();
+    const preferences = useSelector((state: GlobalState) =>
+        getCategory(state, Preferences.FREEMIUM_TRIAL_END_BANNER),
+    );
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const shouldShowBanner = () => {
+        if (!subscription) {
+            return false;
+        }
+
+        if (!limits[1]) {
+            return false;
+        }
```

Comment:
What is this checking? Also, should we do a length check here?

Reply:
the `useGetLimits` hook always returns a list of 2 values - first is the limits themselves, and second is if the limits have actually been received by the front end. This is just so the banner doesn't show before we know that the limits have been received